### ISSUE TITLE
Rule blocked: CORERULES-201

### ISSUE THREAD

**Issue by gerrycampion** [cdisc-org/cdisc-rules-engine#523]
**Links to related JIRA Tickets**
- https://jira.cdisc.org/browse/CORERULES-201

**Rule Information**
- **Standard**: SDTMIG
- **Rule ID**: CG0325
- **Rule Description**: The combination of TESTRL, TEENRL, and TEDUR must be unique

**Describe the bug**
This rule is trying to use `is_not_unique_set` as follows:

```yaml
Check:
  all:
   - name:
        - TESTRL
        - TEENRL
        - TEDUR
     operator: is_not_unique_set
```

There are 2 issues with this operator:
- When one of the variable is missing from the dataset, we get a Variable missing error. Instead, it should attempt `is_not_unique_set` on all of the existing variables.
- When all variables are present, we are getting the following unknown exception:
"Transform function invalid for data types"

The rule and test data are available in prod and attached here:
[Rule (2).zip](https://github.com/cdisc-org/cdisc-rules-engine/files/12412590/Rule.2.zip)

**Comment by nhaydel** [comment]
Name should just be a single value. The value key should have the remaining values in the set

**Comment by gerrycampion** [comment]
Okay. It is a strange way to write the logic for this operator.

For part 1,
Like you said, this works because the name var exists:

```yaml
    - name: TESTRL
      operator: is_not_unique_set
      value:
        - TEENRL
        - TEDUR
```

But this does not work:

```yaml
    - name: TEENRL
      operator: is_not_unique_set
      value:
        - TESTRL
        - TEDUR
```

**Comment by ab3263266** [comment]
@gerrycampion 
How should we address this bug? Do you want it to function in both ways, or should we modify the operation's behavior? To me, it makes sense that if the variable in 'name' is not present in the data, it will skip this rule.

**Comment by gerrycampion** [comment]
@ab3263266 
I gave this some more thought. I think we can actually close this ticket, because there is a way to write this rule using the current code:

```yaml
Check:
  any:
    - name: TEENRL
      operator: is_not_unique_set
      value:
        - TESTRL
        - TEDUR
    - name: TESTRL
      operator: is_not_unique_set
      value:
        - TEENRL
        - TEDUR
    - name: TEDUR
      operator: is_not_unique_set
      value:
        - TESTRL
        - TEENRL
```
